SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc

```diff
@@ -553,6 +552,31 @@ void HcalTriggerPrimitiveAlgo::analyzeQIE11(IntegerCaloSamples& samples,
   IntegerCaloSamples output(samples.id(), tpSamples);
   output.setPresamples(tpPresamples);
 
+  // Based on the |ieta| of the sample, retrieve the correct region "coded" veto threshold
+  // where two of the possible values have special meaning
+  unsigned int codedVetoThreshold = codedVetoThresholds_[theIeta];
+
+  // The var "actualVetoThreshold" will hold the decoded threshold to be used literally in the algo
+  unsigned int actualVetoThreshold;
+  bool applyVetoThreshold = false;
+
+  // Special value to disable vetoing in the PFA1' algo is 0
+  if (codedVetoThreshold > 0) {
+    // Anything in range (1, 2048) inclusive shall activate the veto
+    if (codedVetoThreshold <= 2048) {
+      applyVetoThreshold = true;
+      // Special value to run the veto in PFA1' with no threshold
+      if (codedVetoThreshold == 2048)
+        actualVetoThreshold = 0;
+      // Otherwise take coded threshold as the actual threshold
+      else
+        actualVetoThreshold = codedVetoThreshold;
+    } else {
```

Comment:
Probably simpler (and it avoids letting `actualVetoThreshold` uninitialized):
```suggestion
  unsigned int actualVetoThreshold = codedVetoThreshold;
  // Anything in range (1, 2048) inclusive shall activate the veto
  bool applyVetoThreshold = codedVetoThreshold > 0 && codedVetoThreshold <= 2048;

  if (codedVetoThreshold > 0) {
    if (codedVetoThreshold <= 2048) {
      // Special value to run the veto in PFA1' with no threshold
      if (codedVetoThreshold == 2048)
        actualVetoThreshold = 0;
    } else {
```